api/v1alpha1/openapi.yaml

```diff
@@ -5098,4 +5188,169 @@ components:
           description: Git version of the service.
           type: string
       required:
-        - version
\ No newline at end of file
+        - version
+    Event:
+      type: object
+      required:
+        - kind
+        - apiVersion
+        - id
+        - timestamp
+        - type
+        - source
+        - status
+        - message
+        - severity
+        - correlationId
+        - resourceName
+        - resourceKind
+      properties:
+        apiVersion:
+          type: string
+          description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources.'
+        kind:
+          type: string
+          description: 'Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds.'
+        id:
+          type: integer
+          format: uint64
+          description: Unique identifier for the event.
+        timestamp:
+          type: string
+          format: date-time
+          description: Timestamp when the event occurred.
+        type:
+          type: string
+          description: Type of event.
+          enum:
+            - ResourceCreationSucceeded
+            - ResourceCreationFailed
+            - ResourceUpdateSucceeded
+            - ResourceUpdateFailed
+            - ResourceDeletionSucceeded
+            - ResourceDeletionFailed
+            - DeviceDecommissioningSucceeded
+            - DeviceDecommissioningFailed
+            - EnrollmentRequestApproved
+            - RepositoryAccessible
+            - RepositoryNotAccessible
+            - DeviceThresholdCrossed
+            - DeviceDisconnected
+            - DeviceReconnected
+            - DeviceOnline
+        source:
+          type: string
+          description: The component or service that generated the event.
+          enum:
+            - service-api
+            - service-periodic
+            - service-task
+            - device-agent
+        actorUser:
+          type: string
+          nullable: true
+          description: User who triggered the event (if applicable).
+        actorService:
+          type: string
+          nullable: true
+          description: Automated service or system that triggered the event (if applicable).
+        status:
+          type: string
+          enum: [success, failure, in-progress]
+          description: Current state of the event, indicating whether the action was completed successfully, failed, or is still ongoing.
```

Comment:
This feels potentially overly complex. An event should be emitted after some observation like a snapshot. in-progress makes me think that an event also tracks the lifecycle of the action. is that what we want?